src/databricks_ai_bridge/vector_search_retriever_tool.py

```diff
@@ -78,3 +89,15 @@ def _get_resources(self, index_name: str, embedding_endpoint: str) -> List[Resou
             if embedding_endpoint
             else []
         )
+
+    def _get_tool_name(self) -> str:
+        tool_name = self.tool_name or self.index_name.replace(".", "__")
+
+        # Tool names must match the pattern '^[a-zA-Z0-9_-]+$'."
```

Comment:
should we also check the tool name against this regex. If the tool name they pass in for example has '.' characters, we should throw early instead of letting this propogate and fail when an agent is called.